From f3d1958094c146a91cde33665a19843b5643b1a8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 10 Jul 2018 10:03:07 +0300
Subject: [PATCH] board: habana: add remote_load_firmware script

The script should be installed on the lab host at:
/var/lib/tftpboot/tests
It is used by automatic test frameworks (e.g. tbot, jenkins)
for connecting to the PCIe host, building the habanalabs driver
from the latest sources and installing (insmod) it.
Once installed the driver uploads the firmware (linux) to the ARM
and triggers ARM linux boot.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index 2db7396..27a0be1 100644
--- a/Makefile
+++ b/Makefile
@@ -167,5 +167,5 @@
 .PHONY: dali-asic-tools-install
 dali-asic-tools-install:
 	-cd board/habanalabs/common/ &&					\
-	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
+	sshpass -p123456 scp -r remote_power remote_load_firmware ip_pdu t32 vivado_lab	\
 	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
diff --git a/board/habanalabs/common/remote_load_firmware b/board/habanalabs/common/remote_load_firmware
new file mode 100755
index 0000000..42f6835
--- /dev/null
+++ b/board/habanalabs/common/remote_load_firmware
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+host=$1
+success_str="1000"
+result=~/remote_load_firmware.result
+rm -f $result
+
+for ((i=100; i>0; i--)); do
+	echo $i
+	sleep 1;
+	ping -c 1 -W 1 $host
+	if [ $? -eq 0 ]; then
+		ssh -t $host bash -ic 'hl_repo_sync' | tee -a $result
+		pipestat=${PIPESTATUS[0]}
+		echo ssh repo sync status: $pipestat
+		[ $pipestat -ne 0 ] && continue;
+
+		ssh -t $host bash -ic 'bi_habanalabs' | tee -a $result
+		success=`grep -cE "^$success_str" $result`
+		if [ $success -ge 1 ]; then
+			echo done
+			exit 0;
+		fi
+	fi
+done
+
+echo error
